src/shared/fstab-util.c

```diff
@@ -104,8 +104,14 @@ static int fstab_is_same_node(const char *what_fstab, const char *path) {
         if (path_equal(node, path))
                 return true;
 
-        if (is_device_path(path) && is_device_path(node))
+        if (is_device_path(path) && is_device_path(node)) {
+                /* If the original path to compare with is unavailable, return recognizable error. It's very likely
+                 * that udev hasn't created any devnode symlinks just yet, i.e. we cannot do reasonable comparison. */
+                if (access(path, F_OK) < 0)
```

Comment:
Hmm, here `path` may be something like `/dev/sda3`, and `node` may be something like `/dev/disk/by-path/xxxxxx`. Why only `path` is checked here?

Reply:
Well, we can't possibly cover all these cases. We need to ignore ENOENT in the caller since there might be devices actually missing/expected to be missing (with nofail/noauto/automount/...).

I'd say utilizing gpt-auto, which is supposed to be "portable", in conjunction with non-persistent device names in fstab is too niche that there's little value in supporting really.

Reply:
And path here is passed by gpt-auto, which is known to be by-diskseq/ links.

Reply:
> in conjunction with non-persistent device names in fstab is too niche that there's little value in supporting really

Yes. But then, why not to check `access(node, F_OK)` ? If it does not exist, that _also_ suggests udevd may not be started yet.

Reply:
Thanks. The updated comments sounds nice.